app/main/lib/image_hash.py

```diff
@@ -20,14 +26,41 @@ def compute_ahash(im):
     :returns: Imagehash.ImageHash
   """
   return imagehash.average_hash(ensure_pil(im))
-
 def compute_phash(im):
   """Compute perceptual hash using ImageHash library
     :param im: Numpy.ndarray
     :returns: Imagehash.ImageHash
   """
   return imagehash.phash(ensure_pil(im))
 
+def compute_pdq(im):
+  """Compute perceptual hash using ImageHash library
+    :param im: Numpy.ndarray
+    :returns: Imagehash.ImageHash
+  """
+  with io.BytesIO() as output:
+    ensure_pil(im).save(output, format="GIF")
```

Comment:
Let's either guess the format based on the file extension in the URL , or easier, let's save as TIF format into the buffer, which will not have any compression artefacts/degradation

Reply:
I've updated this to TIF but it still needs to be tested